ext/standard/basic_functions.c

```diff
@@ -3202,7 +3202,7 @@ static int user_shutdown_function_call(zval *zv) /* {{{ */
 	if (!zend_is_callable(&shutdown_function_entry->arguments[0], 0, NULL)) {
 		zend_string *function_name
 			= zend_get_callable_name(&shutdown_function_entry->arguments[0]);
-		php_error(E_WARNING, "(Registered shutdown functions) Unable to call %s() - function does not exist", ZSTR_VAL(function_name));
+		zend_value_error("(Registered shutdown functions) Unable to call %s() - function does not exist", ZSTR_VAL(function_name));
```

Comment:
A potential problem here is that this warnings is triggered during shutdown, so there is actually no way to catch the exception. Maybe that's okay though, I'm not sure.